accelerator/common_core.c

```diff
@@ -99,34 +100,46 @@ int ta_get_transaction_object(const iota_client_service_t* const service,
   }
 
   retcode_t ret = RC_OK;
-  flex_trit_t* tx_trits;
+  flex_trit_t* tx_trits = NULL;
   flex_trit_t hash_trits[NUM_TRITS_HASH];
 
   // get raw transaction data of transaction hashes
+  cache_t* cache = cache_init();
   get_trytes_req_t* get_trytes_req = get_trytes_req_new();
   get_trytes_res_t* get_trytes_res = get_trytes_res_new();
-  if (get_trytes_req == NULL || get_trytes_res == NULL) {
+  if (cache == NULL || get_trytes_req == NULL || get_trytes_res == NULL) {
     goto done;
   }
 
-  flex_trits_from_trytes(hash_trits, NUM_TRITS_HASH, (const tryte_t*)req,
-                         NUM_TRYTES_HASH, NUM_TRYTES_HASH);
-  hash243_queue_push(&get_trytes_req->hashes, hash_trits);
-  ret = iota_client_get_trytes(service, get_trytes_req, get_trytes_res);
-  if (ret) {
-    goto done;
-  }
+  // get in cache first, then get from full node if no result in cache
+  char cache_value[FLEX_TRIT_SIZE_8019 + 1] = {0};
+  ret = cache_get(cache, req, cache_value);
+  if (!ret) {
+    cache_value[FLEX_TRIT_SIZE_8019] = '\0';
```

Comment:
I dislike the way you manipulated with null terminator. Is it really necessary to do?

Reply:
`transaction_t` operations would choose certain length. Looks like it's not necessary.

Reply:
For the sake of security considerations, most operations shall be valid with deterministic length. In other words, we would not hereby depend on null terminator.